radix-transactions/src/validation/transaction_validator.rs

```diff
@@ -1,74 +1,45 @@
-use crate::internal_prelude::*;
-
-pub trait TransactionValidator {
-    type Prepared: TransactionPayloadPreparable;
-    type Validated;
-
-    fn prepare_from_raw(
-        &self,
-        raw: &<Self::Prepared as TransactionPayloadPreparable>::Raw,
-    ) -> Result<Self::Prepared, TransactionValidationError> {
-        self.prepare_from_payload_bytes(raw.as_slice())
-    }
-
-    fn prepare_from_payload_bytes(
-        &self,
-        raw_payload_bytes: &[u8],
-    ) -> Result<Self::Prepared, TransactionValidationError> {
-        if raw_payload_bytes.len() > self.max_payload_length() {
-            return Err(TransactionValidationError::TransactionTooLarge);
-        }
-
-        Ok(Self::Prepared::prepare_from_payload(raw_payload_bytes)?)
-    }
-
-    fn validate_from_raw(
-        &self,
-        raw: &<Self::Prepared as TransactionPayloadPreparable>::Raw,
-    ) -> Result<Self::Validated, TransactionValidationError> {
-        self.validate_from_payload_bytes(raw.as_slice())
-    }
+use radix_substate_store_interface::interface::SubstateDatabase;
 
-    fn validate_from_payload_bytes(
-        &self,
-        payload_bytes: &[u8],
-    ) -> Result<Self::Validated, TransactionValidationError> {
-        let prepared = self.prepare_from_payload_bytes(payload_bytes)?;
-        self.validate(prepared)
-    }
-
-    fn max_payload_length(&self) -> usize;
+use crate::internal_prelude::*;
 
-    fn validate(
-        &self,
-        transaction: Self::Prepared,
-    ) -> Result<Self::Validated, TransactionValidationError>;
+pub trait TransactionPreparer {
+    fn preparation_settings(&self) -> &PreparationSettings;
 }
 
 #[derive(Debug, Copy, Clone, PartialEq, Eq)]
 pub struct ValidationConfig {
     pub network_id: u8,
-    pub max_notarized_payload_size: usize,
+    pub max_signatures_per_intent: usize,
     pub min_tip_percentage: u16,
     pub max_tip_percentage: u16,
     pub max_epoch_range: u64,
     pub message_validation: MessageValidationConfig,
+    pub preparation_settings: PreparationSettings,
 }
 
 impl ValidationConfig {
-    pub fn default(network_id: u8) -> Self {
+    pub fn latest(network_definition: &NetworkDefinition) -> Self {
+        Self::babylon(network_definition.id)
+    }
+
+    pub fn babylon(network_id: u8) -> Self {
         Self {
             network_id,
-            max_notarized_payload_size: MAX_TRANSACTION_SIZE,
+            max_signatures_per_intent: 16,
```

Comment:
Max payload size has now been moved to preparation, as it makes more sense to be there - otherwise we can miss it by accident (!)